Bottom: c5c29aa5449ded15c53eb4c656bc32382f37bd24
Top:    236f2c01beffdeae3632c5b8b6ae4188e602d5b2
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-03-27 02:12:27 +0100

hid/gtk: Ask for polygons before masking as well as after

The GL renderer will need both, so add both now. Teach the GDK renderer
to the ignore gui->use_mask (HID_MASK_BEFORE) calls it will now get.


---

diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index 52d9990..8213bd0 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -270,8 +270,8 @@ ghid_use_mask (int use_it)
       break;
 
     case HID_MASK_BEFORE:
-      printf ("gtk doesn't support mask_before!\n");
-      abort ();
+      /* Ignore, this renderer doesn't use it */
+      return;
 
     case HID_MASK_CLEAR:
       if (!gport->mask)
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 28c10bc..4f16d8c 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -979,7 +979,7 @@ HID ghid_hid = {
   1,				/* gui */
   0,				/* printer */
   0,				/* exporter */
-  0,				/* poly before */
+  1,				/* poly before */
   1,				/* poly after */
   0,				/* poly dicer */
